src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -314,15 +315,7 @@ bool AsyncRPCOperation_sendmany::main_impl() {
         // If there is a single zaddr and there are coinbase utxos, change goes to the zaddr.
         if (change > 0) {
             if (isSingleZaddrOutput && selectedUTXOCoinbase) {
-                std::string address = std::get<0>(zOutputsDeque.front());   
-                SendManyRecipient smr(address, change, std::string());
-                zOutputsDeque.push_back(smr);
-
-                LogPrint("zrpc", "%s: change from coinbase utxo is also sent to the recipient (amount=%s)\n",
-                        getId().substr(0, 10),
-                        FormatMoney(change, false)
-                        );
-
+                throw JSONRPCError(RPC_WALLET_ERROR, strprintf("Change %ld not allowed. When protecting coinbase funds, the wallet does not allow any change as there is currently no way to specify a change address in z_sendmany.", change));
             } else if (!isSingleZaddrOutput && selectedUTXOCoinbase) {
                 // This should not happen and is not allowed
                 assert(false);
```

Comment:
I think it would be slightly clearer to write it as:
```
if (selectedUTXOCoinbase) {
    assert(isSingleZaddrOutput);
    throw JSONRPCError(RPC_WALLET_ERROR, strprintf(
        "Change %ld not allowed. When protecting coinbase funds, the wallet does not allow any change as there is currently no way to specify a change address in z_sendmany.", change));
} else {
```

Also, is the comment two lines below correct? It says "if there is a single zaddr": if this condition is necessary, then it should be asserted, and if not then the comment should not mention it.